manifests/source.pp

```diff
@@ -131,6 +134,12 @@
     }
   }
 
+  $repos.split(' ').each | $repo | {
+    if $repo !~ Regexp(/^(main|contrib|non-free|universe|multiverse|restricted)$/) {
+      fail('repos parameter must be set to a valid component name, or a comma-separated list of valid component names')
```

Comment:
Because of the `split(' ')` above, I think you mean:
```suggestion
      fail('repos parameter must be set to a valid component name, or a space-separated list of valid component names')
```

But maybe the data type should be changed to allow e.g. `Pattern[\A[[:alnum:]]]+\z]` (or an array of this)

Reply:
Thanks for the suggestion @smortex, I have adhere with regex approach.